src/algebra/star/subalgebra.lean

```diff
@@ -225,3 +248,353 @@ set.centralizer_subset h
+      exacts [hcomm _ ha _ hb, star_star b ▸ hcomm_star _ ha _ hb,
+        star_star a ▸ (hcomm_star _ hb _ ha).symm,
+        by simpa only [star_mul, star_star] using congr_arg star (hcomm _ hb _ ha)],
+    end,
+    exact congr_arg coe (mul_comm (⟨x, hx⟩ : algebra.adjoin R (s ∪ star s)) ⟨y, hy⟩),
+  end,
+  ..(adjoin R s).to_subalgebra.to_semiring }
+
+/-- If all elements of `s : set A` commute pairwise and also commute pairwise with elements of
+`star s`, then `star_subalgebra.adjoin R s` is commutative. See note [reducible non-instances]. -/
+@[reducible]
+def adjoin_comm_ring_of_comm (R : Type u) {A : Type v} [comm_ring R] [star_ring R]
+  [ring A] [algebra R A] [star_ring A] [star_module R A] {s : set A}
+  (hcomm : ∀ (a : A), a ∈ s → ∀ (b : A), b ∈ s → a * b = b * a)
+  (hcomm_star : ∀ (a : A), a ∈ s → ∀ (b : A), b ∈ s → a * star b = star b * a) :
+comm_ring (adjoin R s) :=
+{ ..star_subalgebra.adjoin_comm_semiring_of_comm R hcomm hcomm_star,
+  ..(adjoin R s).to_subalgebra.to_ring }
+
+/-- The star subalgebra `star_subalgebra.adjoin R {x}` generated by a single `x : A` is commutative
+if `x` is normal. -/
+instance adjoin_comm_semiring_of_is_star_normal (x : A) [is_star_normal x] :
+  comm_semiring (adjoin R ({x} : set A)) :=
+adjoin_comm_semiring_of_comm R
+  (λ a ha b hb, by { rw [set.mem_singleton_iff] at ha hb, rw [ha, hb] })
+  (λ a ha b hb,
+    by { rw [set.mem_singleton_iff] at ha hb, simpa only [ha, hb] using (star_comm_self' x).symm })
+
+/-- The star subalgebra `star_subalgebra.adjoin R {x}` generated by a single `x : A` is commutative
+if `x` is normal. -/
+instance adjoin_comm_ring_of_is_star_normal (R : Type u) {A : Type v} [comm_ring R] [star_ring R]
+  [ring A] [algebra R A] [star_ring A] [star_module R A]  (x : A) [is_star_normal x] :
+  comm_ring (adjoin R ({x} : set A)) :=
+{ mul_comm := mul_comm, ..(adjoin R ({x} : set A)).to_subalgebra.to_ring }
+
+end star_subalgebra
+
+namespace star_alg_hom
+
+variables {F R A B : Type*} [comm_semiring R] [star_ring R]
+variables [semiring A] [algebra R A] [star_ring A] [star_module R A]
+variables [semiring B] [algebra R B] [star_ring B]
+
+open star_subalgebra
+
+lemma ext_adjoin {s : set A} [star_alg_hom_class F R (adjoin R s) B] {f g : F}
+  (h : ∀ x : adjoin R s, (x : A) ∈ s → f x = g x) : f = g :=
+begin
+  refine fun_like.ext f g (λ a, adjoin_induction' a (λ x hx, _) (λ r, _) (λ x y hx hy, _)
+    (λ x y hx hy, _) (λ x hx, _)),
+  { exact h ⟨x, subset_adjoin R s hx⟩ hx },
+  { simp only [alg_hom_class.commutes] },
+  { rw [map_add, map_add, hx, hy] },
+  { rw [map_mul, map_mul, hx, hy] },
+  { rw [map_star, map_star, hx] },
+end
+
+lemma ext_adjoin_singleton {a : A} [star_alg_hom_class F R (adjoin R ({a} : set A)) B] {f g : F}
+  (h : f ⟨a, self_mem_adjoin_singleton R a⟩ = g ⟨a, self_mem_adjoin_singleton R a⟩) : f = g :=
+ext_adjoin $ λ x hx, (show x = ⟨a, self_mem_adjoin_singleton R a⟩,
+  from subtype.ext $ set.mem_singleton_iff.mp hx).symm ▸ h
```

Comment:
I don't really like the way these two lemmas are stated. I guess you have good reasons to have them stated using subtypes, but could you at least prove the regular version first? Like [alg_hom.ext_of_adjoin_eq_top](https://leanprover-community.github.io/mathlib_docs/ring_theory/adjoin/basic.html#alg_hom.ext_of_adjoin_eq_top) or [linear_map.eq_on_span](https://leanprover-community.github.io/mathlib_docs/linear_algebra/span.html#linear_map.eq_on_span)

Reply:
I have added the regular version and a few supporting bits.

About the subtype version: when establishing the continuous functional calculus, a key step is showing that, if `C⋆(a)` denotes `(adjoin R {a}).topological_closure`, then `character_space ℂ C⋆(a)` is homeomorphic to `spectrum ℂ a`, and consequently there is a `star_alg_equiv` between continuous functions on these spaces.

Now, to construct the homeomorphism, one simply maps `φ : character ℂ C⋆(a)` to `φ a` (in other words, this is evaluation of the Gelfand transform of `a` at `φ`). This map is continuous (because evaluation is), and surjective (because of `spectrum.gelfand_transform_eq`), and since these are compact Hausdorff spaces, if this map is injective, then it will be a homeomorphism. Note that `character_space ℂ C⋆(a)` has a `star_alg_hom_class` instance. The point is then that if `φ a = ψ a`, then these must agree on `adjoin R {a}`, and hence (because they are also continuous) on the closure also, which is `C⋆(a)`, so `φ = ψ`.

A few notes about this:
1. we really need to consider `C⋆(a)`, there is no avoiding it, it's simply the correct object.
2. the domain of `φ` and `ψ` therefore really is the subtype `C⋆(a)`.
3. `adjoin R {a} : star_subalgebra R A`, it is not inherently a star subalgebra of `C⋆(a)`, so things are a bit more complicated than just applying `adjoin_le_equalizer`.
4. To circumvent the issue in (3), I show that `φ` and `ψ` composed with the `star_subalgebra.inclusion` map from `adjoin R {a}` to `ℂ` are equal, and the range of this inclusion map is dense in `C⋆(a)` (so again, by continuity `φ = ψ`).

That's probably far more than you wanted to know, but I'm already avoiding some other pesky and nontrivial details.

Reply:
Thanks for the explanation! I was thinking of something like this indeed, but it's nice to see you have a very clear idea of what you need!